client/settings/express-checkout-settings/woopay-preview.js

```diff
@@ -105,9 +109,62 @@ const CheckoutButton = ( { height } ) => {
 	);
 };
 
+/**
+ * Sanitizes HTML for the preview.
+ *
+ * @param {string} input The HTML to sanitize.
+ * @return {string} The sanitized HTML.
+ */
+function sanitizeHtmlForPreview( input ) {
+	return input.replace( /<\/?([a-zA-Z]+)[^>]*>/g, function (
+		fullMatch,
+		tagName
+	) {
+		tagName = tagName.toLowerCase();
+		const allowedTags = [ 'a', 'em', 'strong', 'b', 'i' ];
+		// Only allow allowedTags.
+		if ( ! allowedTags.includes( tagName ) ) {
+			return '';
+		}
+
+		// 'a' tags are converted to 'span' tags with a class, in the preview.
+		if ( tagName === 'a' ) {
+			if ( fullMatch.startsWith( '</' ) ) {
+				return `</span>`;
+			}
+
+			return `<span class="preview-layout__shortcode-link">`;
+		}
+
+		// Remaining tags are stripped of attributes, in the preview.
+		if ( fullMatch.startsWith( '</' ) ) {
+			return `</${ tagName }>`;
+		}
+
+		return `<${ tagName }>`;
+	} );
+}
+
 export default ( { storeName, storeLogo, customMessage, ...props } ) => {
 	const { style, ...restProps } = props;
-	const trimmedCustomMessage = ( customMessage || '' ).trim();
+
+	const preparedCustomMessage = useMemo( () => {
+		let rawCustomMessage = ( customMessage || '' ).trim();
+
+		if ( rawCustomMessage ) {
+			rawCustomMessage = sanitizeHtmlForPreview( rawCustomMessage );
+			rawCustomMessage = rawCustomMessage.replace(
+				/\[(terms|terms_of_service_link)\]/g,
+				'<span class="preview-layout__shortcode-link">Terms of Service</span>'
+			);
+			rawCustomMessage = rawCustomMessage.replace(
+				/\[(privacy_policy|privacy_policy_link)\]/g,
```

Comment:
We can optimize the regular expressions like the following:

```suggestion
				/\[terms(_of_service_link)*\]/g,
				'<span class="preview-layout__shortcode-link">Terms of Service</span>'
			);
			rawCustomMessage = rawCustomMessage.replace(
				/\[privacy_policy(_link)*\]/g,
```

Reply:
Good call. I like that your version is less verbose. 

But in this case, having the entire term appear will make it easier to find when doing a global search, which I anticipate we will need to do in the future when we phase out the old shortcodes (e.g. `[terms_of_service_link]` and `[privacy_policy_link]`).